pycfmodel/model/types.py

```diff
@@ -37,6 +36,33 @@ def validate(cls, value: NetworkType) -> IPv6Network:
         return IPv6Network(value, False)
 
 
+class SemiStrictBool(int):
+    """
+    SemiStrictBool to allow for bools which are not type-coerced.
+    """
+
+    @classmethod
+    def __modify_schema__(cls, field_schema: Dict[str, Any]) -> None:
+        field_schema.update(type="boolean")
+
+    @classmethod
+    def __get_validators__(cls):
+        yield cls.validate
+
+    @classmethod
+    def validate(cls, value: Any) -> bool:
+        """
+        Ensure that we only allow bools.
+        """
+        if isinstance(value, bool):
+            return value
+
+        if isinstance(value, str) and value.lower() in ("true", "false"):
+            return value == "true"
```

Comment:
What about if value is `"True"` or `"TRUE"`? Do we need to have `value.lower() == "true"`?

Reply:
Made a proposal on how to solve this at [df58525](https://github.com/Skyscanner/pycfmodel/pull/91/commits/df58525c742d2bee42c1c7554ea3e59830697293)
@oscarbc96 can you check that possible solution as well?
Thank you both!